modules/base/rendering/renderablenodedirectionhint.cpp

```diff
@@ -0,0 +1,544 @@
+
+    _invertArrowDirection = p.invert.value_or(_invertArrowDirection);
+    addProperty(_invertArrowDirection);
+
+    _arrowHeadSize = p.arrowHeadSize.value_or(_arrowHeadSize);
+    addProperty(_arrowHeadSize);
+
+    _arrowHeadWidthFactor = p.arrowHeadWidthFactor.value_or(_arrowHeadWidthFactor);
+    addProperty(_arrowHeadWidthFactor);
+
+    _width = p.width.value_or(_width);
+    _width.setExponent(10.f);
+    addProperty(_width);
+
+    _useRelativeLength.onChange([this]() {
+        SceneGraphNode* startNode = sceneGraphNode(_start);
+        if (!startNode) {
+            LERROR(fmt::format("Could not find start node '{}'", _start.value()));
+            return;
+        }
+        const double boundingSphere = startNode->boundingSphere();
+
+        if (!_useRelativeLength) {
+            // Recompute distance (previous value was relative)
+            _length = static_cast<float>(_length * boundingSphere);
+            _length.setExponent(11.f);
+            _length.setMaxValue(1e20f);
+        }
+        else {
+            // Recompute distance (previous value was in meters)
+            if (boundingSphere < std::numeric_limits<double>::epsilon()) {
+                LERROR(fmt::format(
+                    "Start node '{}' has invalid bounding sphere", _start.value()
+                ));
+                return;
+            }
+            _length = static_cast<float>(_length / boundingSphere);
+            _length.setExponent(3.f);
+            _length.setMaxValue(1000.f);
+        }
+    });
+    // @TODO (emmbr, 2022-08-22): make GUI update when min/max value is updated
+
+    _useRelativeLength = p.useRelativeLength.value_or(_useRelativeLength);
+
+    _length = p.length.value_or(_length);
+    addProperty(_length);
+
+    if (!_useRelativeLength) {
+        _length.setExponent(11.f);
+    }
+
+    _useRelativeOffset.onChange([this]() {
+        SceneGraphNode* startNode = sceneGraphNode(_start);
+        if (!startNode) {
+            LERROR(fmt::format("Could not find start node '{}'", _start.value()));
+            return;
+        }
+        const double boundingSphere = startNode->boundingSphere();
+
+        if (!_useRelativeOffset) {
+            // Recompute distance (previous value was relative)
+            _offsetDistance = static_cast<float>(_offsetDistance * boundingSphere);
+            _offsetDistance.setExponent(11.f);
+            _offsetDistance.setMaxValue(1e20f);
+        }
+        else {
+            // Recompute distance (previous value was in meters)
+            if (boundingSphere < std::numeric_limits<double>::epsilon()) {
+                LERROR(fmt::format(
+                    "Start node '{}' has invalid bounding sphere", _start.value()
+                ));
+                return;
+            }
+            _offsetDistance = static_cast<float>(_offsetDistance / boundingSphere);
+            _offsetDistance.setExponent(3.f);
+            _offsetDistance.setMaxValue(1000.f);
+        }
```

Comment:
Can this lambda be merged with the previous one somehow?  Seems to be the `_offsetDistance` vs `_length` is the only difference?